docs/scale/01-about.md

```diff
@@ -0,0 +1,111 @@
+<!--[metadata]>
++++
+title = "Learn the application architecture"
+description = "Try Swarm at scale"
+keywords = ["docker, swarm, scale, voting, application, archiecture"]
+[menu.main]
+parent="scale_swarm"
+weight=-99
++++
+<![end-metadata]-->
+
+# Learn the application architecture
+
+On this page, you learn about the Swarm at scale example.  Make sure you have
+read through [the introduction](index.md) to get an idea of the skills and time
+required first.
+
+## Learn the example back story
+
+Your company is a pet food company that has bought an commercial during the
+Superbowl. The commercial drives viewers to a web survey that asks users to vote
+&ndash; cats or dogs. You are developing the web survey.
+
+Your survey must ensure that millions of people can vote concurrently without
+your website becoming unavailable. You don't need real-time results, a company
+press release announces the results. However, you do need confidence that every
+vote is counted.
+
+
+## Understand the pplication architecture
+
+The voting application is a dockerized microservice application. It uses a
+parallel web frontend that sends jobs to asynchronous background workers. The
+application's design can accommodate arbitrarily large scale. The diagram below
+shows the high level architecture of the application.
+
+![](../images/app-architecture.jpg)
+
+The application is fully dockerized with all services running inside of
+containers.
+
+The frontend consists of an Interlock load balancer with *n* frontend web
+servers and associated queues. The load balancer can handle an arbitrary number
+of web containers behind it (`frontend01`- `frontendN`). The web containers run
+a simple Python Flask application. Each web container accepts votes and queues
+them to a Redis container on the same node. Each web container and Redis queue
+pair operates independently.  
+
+The load balancer together with the independent pairs allows the entire
+application to scale to an arbitrary size as needed to meet demand.
+
+Behind the frontend is a worker tier which runs on separate nodes. This tier:
+
+* scans the Redis containers
+* dequeues votes
+* deduplicates votes to prevent double voting
+* commits the results to a Postgres container running on a separate node
+
+Just like the front end, the worker tier can also scale arbitrarily.
+
+## Swarm Cluster Architecture
+
+To support the application, the design calls for a Swarm cluster with a single
+Swarm manager and four nodes as shown below.
+
+![](../images/swarm-cluster-arch.jpg)
```

Comment:
In the diagram, `N workers` should be changed to `M workers` because `Just like the front end, the worker tier can also scale arbitrarily.` Worker instance count is not tied to frontend instance count.